deploy/helm/sumologic/conf/logs/collector/otelcol/config.yaml

```diff
@@ -1 +1,245 @@
+      ## record.
+      ## Input Body (string): '{"log":"2001-02-03 04:05:06 first line\n","stream":"stdout","time":"2021-11-25T09:59:13.23887954Z"}'
+      ## Output Body (JSON): { "log": "2001-02-03 04:05:06 first line\n", "stream": "stdout" }
+      ## Input Timestamp: _empty_
+      ## Output Timestamp: 2021-11-25 09:59:13.23887954 +0000 UTC
+      - id: parser-docker
+        type: json_parser
+        parse_to: body
+        output: merge-docker-lines
+        timestamp:
+          parse_from: body.time
+          layout: '%Y-%m-%dT%H:%M:%S.%LZ'
+
+      ## merge-docker-lines stitches back together log lines split by Docker logging driver.
+      ## Input Body (JSON): { "log": "2001-02-03 04:05:06 very long li", "stream": "stdout" }
+      ## Input Body (JSON): { "log": "ne that was split by the logging driver\n", "stream": "stdout" }
+      ## Output Body (JSON): { "log": "2001-02-03 04:05:06 very long line that was split by the logging driver\n","stream":"stdout"}
+      - id: merge-docker-lines
+        type: recombine
+        source_identifier: attributes["log.file.path"]
+        output: {{ .Values.sumologic.logs.multiline.enabled | ternary "merge-multiline-logs" "extract-metadata-from-filepath" }}
+        combine_field: body.log
+        combine_with: ""
+        is_last_entry: body.log matches "\n$"
+
+      ## merge-cri-lines stitches back together log lines split by CRI logging drivers.
+      ## Input Body (JSON): { "log": "2001-02-03 04:05:06 very long li", "logtag": "P" }
+      ## Input Body (JSON): { "log": "ne that was split by the logging driver", "logtag": "F" }
+      ## Output Body (JSON): { "log": "2001-02-03 04:05:06 very long line that was split by the logging driver\n", "stream": "stdout" }
+      - id: merge-cri-lines
+        type: recombine
+        source_identifier: attributes["log.file.path"]
+        output: {{ .Values.sumologic.logs.multiline.enabled | ternary "merge-multiline-logs" "extract-metadata-from-filepath" }}
+        combine_field: body.log
+        combine_with: ""
+        is_last_entry: body.logtag == "F"
+        overwrite_with: newest
+
+      ## merge-multiline-logs merges incoming log records into multiline logs.
+      ## Input Body (JSON): { "log": "2001-02-03 04:05:06 first line\n", "stream": "stdout" }
+      ## Input Body (JSON): { "log": "  second line\n", "stream": "stdout" }
+      ## Input Body (JSON): { "log": "  third line\n", "stream": "stdout" }
+      ## Output Body (JSON): { "log": "2001-02-03 04:05:06 first line\n  second line\n  third line\n", "stream": "stdout" }
+{{- if .Values.sumologic.logs.multiline.enabled }}
+      - id: merge-multiline-logs
+        type: recombine
+        output: extract-metadata-from-filepath
+        source_identifier: attributes["log.file.path"]
+        combine_field: body.log
+        combine_with: ""
+        is_first_entry: body.log matches {{ .Values.sumologic.logs.multiline.start_regex | quote }}
+{{- end }}
```

Comment:
I would reconsider supporting multiple mutltiline detections in this PR as this is common use case for customers. I can also create Issue if we prefer to add it in the future

Reply:
I think that should be added in a separate PR. Sounds like a complicated use case.